commands/addcc.ts

```diff
@@ -0,0 +1,109 @@
+            .setName('number_of_assignments')
+            .setDescription('The number of assignments for the class. Add 0 if there are no assignments.')
+            .setRequired(false)
+    );
+
+export const addCc: ICommand = {
+    config: slashCommandBuilder,
+    execute: async (interaction: CommandInteraction) => {
+        await interaction.deferReply();
+        const { channel, options } = interaction;
+        const channelIDs = options.getString('channel');
+        const roleID = options.getRole('role').id;
+        const classTitle = options.getString('title');
+        const classCode = options.getString('class_code');
+        const imageUrl = options.getString('image_url');
+        const type = options.getString('type');
+        const numberOfAssignments = options.getInteger('number_of_assignments') || 0;
+
+        const validateChannel = async (hwChannel: GuildBasedChannel, channelID: string): Promise<boolean> => {
+            const validChannels = {
+                hw: ['GUILD_TEXT', 'GUILD_PUBLIC_THREAD', 'GUILD_PRIVATE_THREAD'],
+                vc: ['GUILD_VOICE'],
+                full_class: ['GUILD_VOICE']
+            };
+            if (!hwChannel) {
+                interaction.followUp(`${channelID} is not a valid channel id`);
+                return false;
+            }
+            if (!validChannels[type].includes(hwChannel.type)) {
+                interaction.followUp(
+                    `Channel type ${hwChannel.type} is not valid for ${type} format. Valid formats are: ${validChannels[
+                        type
+                    ].join(', ')}`
+                );
+                return false;
+            }
+            if (['hw'].includes(type)) {
+                const addedChannelCorrectly = await addHomeworkChannel(hwChannel.id, interaction, classCode);
+                if (!addedChannelCorrectly) {
+                    return false;
+                }
+                channel.send(`Added channel ${hwChannel} (${hwChannel.id}) as a Homework channel`);
+            }
+            return true;
+        };
+
+        if (type === 'hw' && numberOfAssignments === 0) {
+            await interaction.followUp(
+                'For a club, the number of assignments should be greater than 0. <a:shookysad:949689086665437184>'
+            );
+            return;
+        }
+
+        const allChannelIDs = channelIDs.split(' ');
+        const validated = await Promise.all(
+            allChannelIDs.map((hwChannelID) => {
+                const hwChannel = channel.guild.channels.cache.get(hwChannelID);
+                return validateChannel(hwChannel, hwChannelID);
+            })
+        );
+        if (!validated) {
+            return;
+        }
+
+        const classChannel = channel.guild.channels.cache.get(allChannelIDs[0]);
+        const sID = classChannel.guild.id;
+        const result = await getClassCodeByRoleID(roleID);
+        if (result && result.classCode !== classCode) {
```

Comment:
`getClassCodeByRoleID` returns an empty dictionary if no results are found, so the first validation in the if doesn't stop it from executing the if block. Maybe change to this?

```suggestion
        if (result.classCode && result.classCode !== classCode) {
```